src/main/java/net/revelc/code/apilyzer/maven/plugin/AnalyzeMojo.java

```diff
@@ -176,6 +183,72 @@
   @Parameter(alias = "ignoreProblems", property = "apilyzer.ignoreProblems", defaultValue = "false")
   private boolean ignoreProblems;
 
+  /**
+   * This option enables including classes in your public API definition based on class level
+   * annotations. This option takes one or more regular expression. Annotations are discovered using
+   * reflection, so annotations scoped to compile may not be seen. For each regular expression
+   * {@link String#matches(String)} is called on the output of {@link Annotation#toString()}. If any
+   * annotation matches any regular expression and it does not match any exclusion, then its
+   * included as an API type.
+   * 
+   * <p>
+   * This section of the configuration is ORed with the {@code <includes>} section. So if a class
+   * matches something in either section (and its not excluded), then its included in the API
+   * definition.
+   * 
+   * <p>
+   * This section has the same behavior with inner classes as {@code <includes>}.
+   * 
+   * <p>
+   * The following example shows using annotations to analyze Hadoop's API. When trying to analyze a
```

Comment:
Would prefer to have the Hadoop example in the generated site documentation, rather than in the javadoc. This won't look that great in the generated HTML table for the maven plugin site. Can change later.

Reply:
I have not looked at the generated site docs, need to do that.

Reply:
If we add it to the site docs, would be nice to add it in addition to whats  is there. I Think this is a complex example, so it would be nice to still have a simpler example.